picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (C) 2014 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.picasso3
+
+import android.content.ContentResolver
+import android.content.ContentUris
+import android.content.Context
+import android.database.Cursor
+import android.graphics.Bitmap
+import android.net.Uri
+import android.provider.MediaStore
+import android.provider.MediaStore.Video
+import com.squareup.picasso3.BitmapUtils.calculateInSampleSize
+import com.squareup.picasso3.BitmapUtils.createBitmapOptions
+import com.squareup.picasso3.BitmapUtils.decodeStream
+import com.squareup.picasso3.Picasso.LoadedFrom
+
+internal class MediaStoreRequestHandler(context: Context) : ContentStreamRequestHandler(context) {
+
+  override fun canHandleRequest(data: Request): Boolean {
+    val uri = data.uri
+    return uri != null
+        && ContentResolver.SCHEME_CONTENT == uri.scheme && MediaStore.AUTHORITY == uri.authority
+  }
+
+  override fun load(picasso: Picasso, request: Request, callback: Callback) {
+    var signaledCallback = false
+    try {
+      val contentResolver = context.contentResolver
+      val requestUri = checkNotNull(request.uri, { "request.uri == null" })
```

Comment:
i wonder what's the difference between this and `request.uri!!`?

Reply:
I did not want to just do `!!` given you guys were using the `Utils` stuff here, I can change it to `!!` if you prefer

Reply:
Ah, the difference is NPE vs IllegalStateException:

<img width="777" alt="Screen Shot 2022-01-16 at 9 47 18 PM" src="https://user-images.githubusercontent.com/1868149/149700505-be3f165b-56f7-47e2-9f35-96279e40333f.png">

Let's leave it as is, can always change it later if we change our minds.